sources/editor/Stride.Core.Assets.Editor/Services/UserDocumentationService.cs

```diff
@@ -95,59 +104,142 @@ private bool CacheAssemblyDocumentation([NotNull] Assembly assembly)
                     if (ViewModel.SessionViewModel.Instance.CurrentProject?.Package != null)
                     {
                         var package = ViewModel.SessionViewModel.Instance.CurrentProject.Package;
-                        foreach (var asm in package.LoadedAssemblies)
+
+                        if (package.Container is SolutionProject solutionProject && solutionProject.Type == ProjectType.Executable)
+                        {
+                            Log.Info($"Package {solutionProject.Name} is a solution project. Attempting to cache documentation for dependencies.");
+                            foreach (var dep in solutionProject.DirectDependencies)
+                            {
+                                var docPath = Path.Combine(Path.GetDirectoryName(solutionProject.TargetPath) ?? "", dep.Name + ".xml");
+
+                                CacheAssemblyDocumentationFromPath(dep.Name, docPath);
+                            }
+                        }
+                        else
                         {
-                            var name = asm.Assembly.GetName();
-                            if (name.Name == assemblyName)
+                            foreach (var asm in package.LoadedAssemblies)
                             {
-                                location = asm.Path;
-                                break;
+                                var name = asm.Assembly.GetName();
+                                if (name.Name == assemblyName)
+                                {
+                                    location = asm.Path;
+                                    break;
+                                }
                             }
                         }
```

Comment:
A nitpick if you have time: this function works fine but it's kind of a mess right now; 
Those scopes don't really need to be as deep as they are, ex: you can merge two ``if`` together, return from one other ...
You could swap some of your logic around to avoid calling ``CacheAssemblyDocumentationFromPath(..., location)`` below when ``location`` has not changed, like calling it within the if scope above this instead and move the one below this to an else of the ``if (string.IsNullOrEmpty(location))`` test.

Reply:
I did some superficial work to flatten this logic. I didn't go crazy since it's late, but let me know if you'd like to see more than that.